src/status_im/ui/components/qr_code_viewer/views.cljs

```diff
@@ -6,23 +6,63 @@
             [status-im.ui.components.svg :as svg]
             ["qrcode" :as qr-code-js]))
 
-(defn qr-code [{:keys [size value]}]
-  (let [uri (reagent/atom nil)]
-    (.toString
-     qr-code-js
-     value
-     (bean/->js {:margin 0 :width size})
-     #(reset! uri %2))
+(defn qr-logo [{:keys [size identicon]}]
+  (let [background-size     (* size 0.25)
+        logo-size           (* size 0.2)
+        logo-position       (/ (- size logo-size) 2)
+        background-position (/ (- size background-size) 2)]
+    [svg/g
+     [svg/defs
+      [svg/clip-path {:id "logo-clip"}
+       [svg/rect {:width         logo-size
+                  :height        logo-size
+                  :stroke        "blue"
+                  "stroke-width" 2
+                  :rx            (/ logo-size 2)
+                  :ry            (/ logo-size 2)}]]]
+     [svg/g {:x background-position :y background-position}
+      [svg/rect {:width  background-size
+                 :height background-size
+                 :fill   "white"}]]
+     [svg/g {:x logo-position :y logo-position}
+      [svg/rect {:width        logo-size
+                 :height       logo-size
+                 :rx           (/ logo-size 2)
+                 :ry           (/ logo-size 2)
+                 :fill         "rgba(0,0,0,0)"
+                 :stroke       "rgba(0,0,0,0.1)"
+                 :stroke-width 1}]
+      [svg/image {:width               logo-size
+                  :height              logo-size
+                  :href                identicon
+                  :clip-path           "url(#logo-clip)"
+                  :preserveAspectRatio "xMidYMid slice"}]]]))
+
+(defn qr-code [{:keys [size value identicon]}]
+  (let [path (reagent/atom nil)]
+    (.toString qr-code-js
+               value
+               (bean/->js {:margin 0 :width size :errorCorrectionLevel "H"})
```

Comment:
Can we either have only one key/value pair per line or separate pairs with comma? It is a bit hard to read for me personally

Reply:
I totally agree with you, and I would prefer to have that as a lint rule. We have many places where it's multiple pairs per line, and IMHO it's even harder to read when it's multiple pairs per line and multiple lines for the same map.